services/bundle_analysis/report.py

```diff
@@ -118,25 +118,48 @@ async def initialize_and_save_report(
             db_session.flush()
         return commit_report
 
```

Comment:
Refactor this a little bit, instead of having `_previous_bundle_analysis_report` fetch the commit to use then fetch the BA report, it will call `_get_parent_commit` to fetch the commit. `_get_parent_commit` function will try to select the "premerge commit" if available in the head BA report if not default back to selecting commit.parent.